### PR TITLE
[Storage] [Queue] Fixed next pylint errors in queue storage

### PR DESCRIPTION
**Pull request by weirongw23-msft**
# Description

Fixed all next `pylint` errors in queue storage except `docstring-keyword-should-match-keyword-only`. This is the fix for issue https://github.com/Azure/azure-sdk-for-python/issues/32636.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py [hunk 1]
```diff
@@ -852,7 +852,7 @@ def peek_messages(
             A list of :class:`~azure.storage.queue.QueueMessage` objects. Note that
             next_visible_on and pop_receipt will not be populated as peek does
             not pop the message and can only retrieve already visible messages.
-        :rtype: list(:class:`~azure.storage.queue.QueueMessage`)
+        :rtype: List[QueueMessage]
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:855]: Can you leave this one as-is unless Pylint is complaining about it? The way it was will look better in generated docs.

File: sdk/storage/azure-storage-queue/azure/storage/queue/aio/_queue_client_async.py [hunk 2]
```diff
@@ -851,7 +851,7 @@ async def peek_messages(
             A list of :class:`~azure.storage.queue.QueueMessage` objects. Note that
             next_visible_on and pop_receipt will not be populated as peek does
             not pop the message and can only retrieve already visible messages.
-        :rtype: list(:class:`~azure.storage.queue.QueueMessage`)
+        :rtype: list[QueueMessage]
```
Comment (jalauzon-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/aio/_queue_client_async.py:854]: Same with this, revert this change is Pylint is okay with it.

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py (hunk 1 above)
Comment (weirongw23-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:855]: Looks like `pylint` is complaining about `:class:` part. I will just remove that part so the return type doc will look like this:
`:rtype: List[~azure.storage.queue.QueueMessage]`

Comment (weirongw23-msft) [reply sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client.py:855]: Updating this based on previous discussion, will change it to
`list[~azure.storage.queue.QueueMessage]`
Since `list` is a built-in datatype.

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client_helpers.py [hunk 3]
```diff
@@ -32,7 +32,7 @@ def _parse_url(
         - except in the case of AzureSasCredential, where the conflicting SAS tokens will raise a ValueError.
         If using an instance of AzureNamedKeyCredential, "name" should be the storage account name, and "key"
         should be the storage account key.
-    :paramtype credential: Optional[Union[str, Dict[str, str], AzureNamedKeyCredential, AzureSasCredential, AsyncTokenCredential, TokenCredential]] # pylint: disable=line-too-long
+    :type credential: Optional[Union[str, Dict[str, str], AzureNamedKeyCredential, AzureSasCredential, AsyncTokenCredential, TokenCredential]] # pylint: disable=line-too-long
```
Comment (weirongw23-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_queue_client_helpers.py:35]: As per discussion, moved the `credential` type to a newline to remove unnecessary `pylint: disable=line-too-long`, but this file is an exception since the union types will be too long to fit on one line

File: sdk/storage/azure-storage-queue/azure/storage/queue/_queue_service_client_helpers.py [hunk 4]
```diff
@@ -30,7 +30,7 @@ def _parse_url(
         - except in the case of AzureSasCredential, where the conflicting SAS tokens will raise a ValueError.
         If using an instance of AzureNamedKeyCredential, "name" should be the storage account name, and "key"
         should be the storage account key.
-    :paramtype credential: Optional[Union[str, Dict[str, str], AzureNamedKeyCredential, AzureSasCredential, AsyncTokenCredential, TokenCredential]] # pylint: disable=line-too-long
+    :type credential: Optional[Union[str, Dict[str, str], AzureNamedKeyCredential, AzureSasCredential, AsyncTokenCredential, TokenCredential]] # pylint: disable=line-too-long
```
Comment (weirongw23-msft) [inline sdk/storage/azure-storage-queue/azure/storage/queue/_queue_service_client_helpers.py:33]: Same issue with this file, moved to a newline but will still need `pylint` ignore flag.

**Review by vincenttran-msft — APPROVED** [review]
LGTM, thanks for getting to this and verifying how some things are rendering in the docs 😄

**Review by jalauzon-msft — APPROVED** [review]
Left one reply to a comment, otherwise looks good.
